src/config/reader.rs

```diff
@@ -18,16 +18,17 @@ use crate::runtime::TargetRuntime;
 /// linked extensions to create a ConfigModule.
 pub struct ConfigReader {
     runtime: TargetRuntime,
-    resource_reader: ResourceReader,
+    resource_reader: Arc<ResourceReader>,
```

Comment:
```suggestion
    resource_reader: ResourceReader,
```
No need for an arc here, like TargetRuntime, implement clone on ResourceReader if required.